### ISSUE TITLE
Consulta de datos Mediante API

### ISSUE THREAD

**Issue by fariss2** [fariss2/TFG#3]
Acceder a datos UV y Tª en tiempo real

**Comment by fariss2** [comment]
Radiacion UV y altitud* obtenidas. 
*Ver doc de dudas

**Comment by fariss2** [comment]
obtenidas por web scraping* pregunatr error solicitud

**Comment by fariss2** [comment]
trabajar con promedio de estaciones meteorológicas de cada provincia

**Comment by fariss2** [comment]
@ajcanepa ya trabajo con la api proporcionada en esta web (https://opendata.aemet.es/opendata/api/prediccion/especifica/uvi/{dia}) pongo en el parametro dia el codigo 0 que es el de dia actual pero me devuelve error 404 en Rstudio, en cambio si leo la web de  https://www.aemet.es/es/eltiempo/prediccion/radiacionuv?w=0&datos=det si que me va actualizando los indices segun el dia que ejecute, entonces nose con cual quedarme.
También, cuando hago la primera solicitud  con la api y procedo a leer la url que me devuelve donde se almacenan los indices no me la reconoce.

**Comment by ajcanepa** [comment]
Hola @fariss2 , buenas.
No entiendo al 100% el error que tienes de "Not Found (404)". Entiendo que el error te lo muestra en R (ojo, _R es el lenguaje de programación que estás usando_), solo al conectarte con la API pero no al leer desde la página web.

¿Estás usando algún paquete para conectarte a la API?, has intentado descargar otro producto? para saber si el problema es de ellos (sí puedes descargar precipitaciones, por ejemplo) o es tuyo (no puedes descargar ningún otro producto)....?

De momento quédate con la que vaya funcionando; para ir implementando la Aplicación y así ganas tiempo.

Si usas un paquete, siempre puedes leer en los Issue de ese paquete o bien abrir uno nuevo.

Saludos!

**Comment by fariss2** [comment]
Buenas @ajcanepa, 
Podemos reunirnos el jueves por la mañana si tienes un rato libre ?
Un saludo

**Comment by ajcanepa** [comment]
> Buenas @ajcanepa, Podemos reunirnos el jueves por la mañana si tienes un rato libre ? Un saludo

Sí, pero no tengo mucho tiempo. Podría entre las 13:00 y las 13:40 hrs...

**Comment by fariss2** [comment]
> > Buenas @ajcanepa, Podemos reunirnos el jueves por la mañana si tienes un rato libre ? Un saludo
> 
> Sí, pero no tengo mucho tiempo. Podría entre las 13:00 y las 13:40 hrs...

Vale, nos vemos el jueves

**Comment by fariss2** [comment]
no hay endpoint para obtener Tª diarias

**Comment by fariss2** [comment]
consulta a la AEMET: las temperaturas no se validan en tiempo real

**Comment by ajcanepa** [comment]
> consulta a la AEMET: las temperaturas no se validan en tiempo real

Hola @fariss2 ,
No entendí bien si estos mensajes en el *__hilo del issue__* tambi´en eran para mí o no.
Tampoco entiendo lo de la validación en tiempo real....

He visto en el paquete de [climaemet](https://ropenspain.github.io/climaemet/), que muestran un código con una consulta de temperatura, ¿no te valdría a tí algo similar?

```
## Get recorded extreme values of temperature for a station
data_extremes <- aemet_extremes_clim(station, parameter = "T") # T es de temperature
knitr::kable(head(data_extremes))
```
Ya me dices....

**Comment by fariss2** [comment]
obtencion de datos observacionales de los cuales hay posibilidad de filtrar las variables Tmin Tmax

**Comment by fariss2** [comment]
Faltan temperaturas max y min

**Comment by fariss2** [comment]
Para las defunciones no ha sido api sino el paquete de la INE

**Comment by fariss2** [comment]
Buenos dias @ajcanepa, ya tengo todos los datos solicitados y ya visibles en r pero hay un problema con las temperaturas ya qye me los dan por estaciones, que son 800 en España, y ahora mismo estoy intentando buscar las estaciones por provincias y tampoco me sale nada y quedarne con 54 estaciones estaria perdiendo informacion. le adjunto imagen para que te puedas hacer una idea de lo que tengo, abajo del todo se ve q son 832 estaciones meteorológicas

![Image](https://github.com/user-attachments/assets/7504acd0-3858-4d69-bba9-b1e2932fa005)

**Comment by fariss2** [comment]
la solicitud de estos datos la realice ayer a las 11 de la noche, y me dio datos de hasta las 20h no se como ves eso @ajcanepa

**Comment by ajcanepa** [comment]
Hola @fariss2 ,
En principio el tiempo de retardo no sería un problema; lo bueno es que estás pudiendo consultarlas y et están regresando valores, osea que OK.

No entiendo lo que quedarte con 54 estaciones (de dónde sale ese número?), ni tampoco lo de buscarlas por provincia. Lo más probable es que exista una referencia para la variable `idema` que te permita identificar las que pertenecen a una provincia u a otra. 

Por el contrario y aunque quizás más tedioso, es que puedas construir una lista (`list`) a modo de diccionario, para unificar todos los elementos de `ubi` en la provincia que le toca. (Esto es "_picar fino_" y esperemos no tener que llegar a esto. 

Si tienes las coordenadas de las estaciones, también lo podrñias hacer un poco más automatizado....

**Comment by fariss2** [comment]
datos temp 2023 obtenidos

**Comment by fariss2** [comment]
obtener los uv mensuales y trabajar con provincias en vez de idemas

**Comment by fariss2** [comment]
adquisicion de datos 2023 y api finalizada
